stdlib/public/core/StringCreate.swift

```diff
@@ -79,13 +79,35 @@ extension String {
   ) -> (result: String, repairsMade: Bool) {
     switch validateUTF8(input) {
     case .success(let extraInfo):
-        return (String._uncheckedFromUTF8(
-          input, asciiPreScanResult: extraInfo.isASCII
-        ), false)
+      return (String._uncheckedFromUTF8(
+        input, asciiPreScanResult: extraInfo.isASCII
+      ), false)
     case .error(let initialRange):
         return (repairUTF8(input, firstKnownBrokenRange: initialRange), true)
     }
   }
+  
+  @_effects(releasenone)
+  internal static func _fromUTF8Repairing(
+    uninitializedCapacity capacity: Int,
+    initializingWith initializer: (
+      _ buffer: UnsafeMutableBufferPointer<UInt8>
+    ) throws -> Int
+  ) rethrows -> String {
+    let result = try __StringStorage.create(
+      uninitializedCapacity: capacity,
+      initializingUncheckedUTF8With: initializer)
+    
+    switch validateUTF8(result.codeUnits) {
+    case .success(let info):
+      result.forceSetIsASCII(info.isASCII)
+      result._invariantCheck()
+      return result.asString
+    case .error(let initialRange):
+      //This could be optimized to use excess tail capacity
+      return repairUTF8(result.codeUnits, firstKnownBrokenRange: initialRange)
+    }
+  }
```

Comment:
We should probably come up with a different name or structure. All of the other `_fromUTF8` methods do the smol check first.